tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -166,57 +180,57 @@ def process(self, resources):
         self.client = self.manager.get_client('azure.mgmt.monitor.MonitorManagementClient')
```

Comment:
I think we can create some function before_process() in BaseAction and call it from process().. That way we will do all required initialization without the need to override process()

Reply:
I actually took a look again at what was being set in theses process methods that were overriding the base class and they were all things that should be set in init or in process_resource_set. Refactored out so none of these need to override process anymore.